addons/purchase/models/purchase.py

```diff
@@ -988,6 +988,21 @@ def _make_po_select_supplier(self, suppliers):
         """
         return suppliers[0]
 
+    def _get_po_domain(self):
+        gpo = self.rule_id.group_propagation_option
+        group = (gpo == 'fixed' and self.rule_id.group_id) or \
+                (gpo == 'propagate' and self.group_id) or False
+
+        domain = (
+            ('partner_id', '=', self.id),
```

Comment:
Nope :-) should be partner.id It is fixed in https://github.com/odoo/odoo/pull/17192